quarkus/CONTRIBUTING.md

```diff
@@ -0,0 +1,211 @@
+[pom.xml](server/pom.xml) from this module you'll notice that it is a very simple Quarkus application with the bare minimum configuration
+to actually build and run the server.
+
+As Quarkus application, the server is a [mutable](https://quarkus.io/guides/reaugmentation#how-to-re-augment-a-quarkus-application) application using the 
+[mutable-jar](server/src/main/resources/application.properties) package type. As a mutable application, Keycloak is able to allow
+users to configure some aspects of the server without having to re-build this module, something impractical from a user perspective.
+
+The mutability of the server is directly related to the `build` command. As mentioned before, the build steps from the `deployment` module
+are only run when (re)augmenting the server and when running the `build` command, the server will indicate to Quarkus that the build steps
+should re-run and the recorded bytecode should be updated to reflect any change to the server configuration.
+
+From a Quarkus perspective, the server is also a [Command Mode Application](https://quarkus.io/guides/command-mode-reference) and provides
+a CLI based on [Picocli](https://picocli.info/). As such, there is a single entrypoint that executes the code to execute the CLI and bootstrap the server. This entry point
+is the `org.keycloak.quarkus.runtime.KeycloakMain` class from the `runtime` module.
+
+## Keycloak Distribution
+
+The server distribution is created by build the `dist` module. This module basically consists of packaging the `deployment`, `runtime`,
+and `server` modules artifacts and their dependencies to a `ZIP` or tarball file.
+
+Within this directory you'll find the directory structure of the distribution and what is included in it.
+
+## Running the server in development mode
+
+As a regular Quarkus application, you should be able to run the `server` module in [dev mode](https://quarkus.io/guides/maven-tooling#dev-mode) just like any regular application:
+
+```
+cd server
+mvn clean quarkus:dev -Dquarkus.args="start-dev"
+```
+
+You can set any command or configuration option to the server by setting the `quarkus.args` environment variable.
+
+When running in dev mode, you can benefit from the dev mode capabilities from Quarkus but with some limitations. The main limitations you'll find
+at the moment are:
+
+* Changes are only automatically reflected at runtime if you are changing resources from the `deployment`, `runtime`, and `server` modules. Other modules, such as `keycloak-services` still rely on Hot Swap in Java debuggers to reload classes.
+* There is nothing in the Dev UI related to the server itself, although you can still change some configuration from there.
+* There are some limitations when passing some options when running in dev mode. You should expect more improvements in this area.
```

Comment:
I don't think it makes sense to add them now because we should be fixing some of them:

* The ability to pass certain build options via CLI and have them recognized by the server
* Improvements to Dev UI
* Improvements to run the server using a specific database by spinning up a testcontainer database
* See how we can recognize changes to dependencies and automatic restart the application
* Make sure the experience is at least the same when using Undertow for development

Reply:
np, just felt a bit vague for me. But as you prefer.